### PR TITLE
Change to new C++ based addon interface style

### PR DESCRIPTION
**Pull request by Rechi**
This are changes related to Kodi request xbmc/xbmc#12262

@AlwinEsch please check if the new interface is used correct.

### CODE REVIEW

File: src/Milkdrop2XBMC.cpp [hunk 1]
```diff
@@ -162,96 +145,46 @@ char* WideToUTF8( const wchar_t* WFilename )
...
 {
-	if( !presets || !IsInitialized )
-	{
-		return 0;
-	}
+	if(!IsInitialized )
+		return false;
 
 	while( !g_plugin.m_bPresetListReady )
 	{
```
Comment (AlwinEsch) [inline src/Milkdrop2XBMC.cpp:155]: Just a small question for what is this :-) ? Is a return false not enough?

**Comment by AlwinEsch** [comment]
Thanks a lot @Rechi, only a bit confused with the possible dead loop there on the old parts :-D

File: src/Milkdrop2XBMC.cpp (hunk 1 above)
Comment (Rechi) [reply src/Milkdrop2XBMC.cpp:155]: I don't know what this does. Also I tried to keep as much as possible in the original way to not introduce any bugs.

Comment (afedchin) [reply src/Milkdrop2XBMC.cpp:155]: it's waiting for the completion of loading of preset list which happens in parallel

**Comment by Rechi** [comment]
@AlwinEsch is this ready to get merged?

**Comment by AlwinEsch** [comment]
Ready to merge from my side.

**Pull request merged**